src/mpid/ch4/shm/ipc/gpu/gpu_post.h

```diff
@@ -38,14 +38,10 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_GPU_get_mem_attr(const void *vaddr, MPIDI_IPC
 
 #ifdef MPIDI_CH4_SHM_ENABLE_GPU
```

Comment:
I thought we decided yesterday that this configure-time check was not needed, given that there's already a configure-time check inside MPL.

Reply:
We leave `MPIDI_CH4_SHM_ENABLE_GPU` to control GPU IPC functionality. I also left a note on slack at 11:45pm yesterday:
>I still check MPIDI_CH4_SHM_ENABLE_GPU in MPIDI_GPU_get_mem_attr , for the case where MPIR_ENABLE_GPU is true, but the user wants to only disable IPC support. Then, we need set the IPC_type to none in order to fallback to POSIX

Reply:
In that case, this CVAR is misleading.  It's not enabling/disabling GPU support for SHM.  It's enabling the usage of GPUDirect IPC.  If you want one for this, you'd need to generalize it to include the other GPU features too, such as GPUDirect RDMA.

Reply:
This is a macro generated by `--with-ch4-shmmods=gpu`. The macro was named following the XPMEM one. I think we can adjust them consistently in a separate configure fix PR.

I think defining other configure option or CVAR for GPUDirect RDMA is out of scope of this PR.